LDAR_Sim/model_code/ldar_sim_main.py

```diff
@@ -24,47 +24,88 @@
 import pandas as pd
 import os
+import sys
 import shutil
 import datetime
 import warnings
 import multiprocessing as mp
 from generic_functions import check_ERA5_file
 
+from input_manager import InputManager
+
 if __name__ == '__main__':
     # ------------------------------------------------------------------------------
-    # -----------------------------Global parameters--------------------------------
-    src_dir_path = Path(os.path.dirname(os.path.realpath(__file__)))
-    src_dir = str(src_dir_path)
-    root_dir = str(src_dir_path.parent)
-    wd = os.path.abspath(root_dir) + "/inputs_template/"
-    output_directory = os.path.abspath(root_dir) + "/outputs/"
-    # Programs to compare; Position one should be the reference program (P_ref)
-    program_list = ['P_ref', 'P_dev_continuous', 'P_cont']
-
-# -----------------------------Set up programs----------------------------------
-    programs = []
-
-    warnings.filterwarnings('ignore')    # Temporarily mute warnings
-
-    for p in range(len(program_list)):
-        file = wd + program_list[p] + '.txt'
-        exec(open(file).read())
-        programs.append(eval(program_list[p]))
-
-    n_processes = programs[0]['n_processes']
-    print_from_simulations = programs[0]['print_from_simulations']
-    n_simulations = programs[0]['n_simulations']
-    spin_up = programs[0]['spin_up']
-    ref_program = program_list[0]
-    write_data = programs[0]['write_data']
-
+    # Look for parameter files supplied as arguments - if parameter files are supplied as
+    # arguments, proceed to parse and type check input parameter type with the input manager
+    parameter_filenames = sys.argv[1:]
+
+    # FOR TESTING ONLY!
+    print('WARNING: USING HARD CODED INPUT FILES FOR TESTING ONLY - REMOVE ON LINE 43 OF ldar_sim_main.py')
+    parameter_filenames = [
+        '..//sample_simulations//simple_OGI.yaml',
+        '..//sample_simulations//P_ref.yaml',
+        '..//sample_simulations//OGI.yaml',
+    ]
+
+    if len(parameter_filenames) > 0:
+        print('LDAR-Sim using parameters supplied as arguments')
+        input_manager = InputManager()
+        simulation_parameters = input_manager.read_and_validate_parameters(parameter_filenames)
+
+        # Assign appropriate local variables to match older way of inputting parameters
+        wd = simulation_parameters['wd']
+        output_directory = simulation_parameters['output_directory']
+        programs = simulation_parameters['programs']
+        n_processes = simulation_parameters['n_processes']
+        print_from_simulations = simulation_parameters['print_from_simulations']
+        n_simulations = simulation_parameters['n_simulations']
+        spin_up = simulation_parameters['spin_up']
+        ref_program = simulation_parameters['reference_program']
+        write_data = simulation_parameters['write_data']
+        weather_file = simulation_parameters['weather_file']
+        start_year = simulation_parameters['start_year']
```

Comment:
Should we skip this and just use the simulation_parameters['key'] ?

Reply:
This is to enable precise compatiblity with the way it was before so everyone can keep using the 'P_*' files like they used to lower down in this file. I'm not too picky, whatever.

Reply:
Leave in for now, its easy to remove

Reply:
This should be removed eventually but I want to see people using the new parameter apparatus and be happy with it.